src/mappings/shared/token/setMetadata.ts

```diff
@@ -26,3 +26,35 @@ export async function setMetadataHandler(context: Context, collection: CE, nft:
   const existingToken = await getOptional<TE>(context.store, TE, generateTokenId(collection.id, nftMedia))
   return await (existingToken ? tokenAPI.addNftToToken(nft, existingToken) : tokenAPI.create(collection, nft))
 }
+
+export async function setMetadataOnCollectionHandler(context: Context, collection: CE): Promise<TE[] | undefined> {
```

Comment:
This is O(n) writes to DB. 
It would slow the indexer a lot. :)

Reply:
thanks vik
how about now?